File: src/rainbow/inner/coordinate/point/PointDouble.java
package rainbow.inner.coordinate.point;

/**
 * 用于表示屏幕上的位置
 *
 * @author Rainbow Yang
 */
public class PointDouble {
    private double x;
    private double y;

    public PointDouble(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public void setX(double x) {
        this.x = x;
    }

    public void setY(double y) {
        this.y = y;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    @Override
    public String toString() {
        return "PointDouble{" +
                "x=" + x +
                ", y=" + y +
                '}';
    }
}

File: src/main/kotlin/rainbow/inner/coordinate/point/PointDouble.kt
package rainbow.inner.coordinate.point

/**
 * 用于表示屏幕上的位置
 *
 * @author Rainbow Yang
 */
class PointDouble(val x: Double, val y: Double) {

    constructor(x: Int, y: Int) : this(x.toDouble(), y.toDouble())

    fun spin(angle: Double): PointDouble {
        return PointDouble(x * Math.cos(angle) - y * Math.sin(angle), x * Math.sin(angle) + y * Math.cos(angle))
    }

    override fun toString(): String {
        return "PointDouble(x=$x, y=$y)"
    }

}